cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -96,6 +98,64 @@ class CwlWorkflowWomSpec extends FlatSpec with Matchers {
     }
   }
 
+  behavior of "A decoded CWL 3step"
+
+  /**
+    * Runtime version of `refineMV[ECMAScriptExpression]("$(foo)")` as I cannot summon the dark magic for compile
+    * time checks.
+    *
+    * Compiler output after `import eu.timepit.refined._`:
+    *
+    * Error:(_, _) could not find implicit value for parameter v: eu.timepit.refined.api.Validate[String,wdl4s.cwl.ECMAScriptExpression]
```

Comment:
is this perhaps how it tells you that it didn't match?

Reply:
The same string matches at runtime. ☹️ Not sure what I'm doing wrong.

Reply:
Ok I think I figured it out.  The `MatchesRegex` section of `ECMAScriptExpression` is the "predicate", so when you are declaring w/ `refineMV`, you use _only_ that.  

observe:
```scala
scala> val foo = refineMV[MatchesRegex[W.`"""\\$\\([^)]*\\)"""`.T]]("$(foo)")
foo: eu.timepit.refined.api.Refined[String,eu.timepit.refined.string.MatchesRegex[String("\\$\\([^)]*\\)")]] = $(foo)

scala> type Y = MatchesRegex[W.`"""\\$\\([^)]*\\)"""`.T]
defined type alias Y

scala> val foo = refineMV[Y]("$(foo)")
foo: eu.timepit.refined.api.Refined[String,Y] = $(foo)

scala> val x: ECMAScriptExpression = foo
x: wdl4s.cwl.ECMAScriptExpression = $(foo)
```

Reply:
cool, and it looks like that type can be pulled out so it can be used at compile and runtime.